widget_driver_generator/lib/src/utils/providable_variable_code_generator.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:widget_driver_generator/src/utils/code_writer.dart';
+import 'package:widget_driver_generator/src/utils/providable_field.dart';
+import 'package:widget_driver_generator/src/utils/source_code_generator.dart';
+
+/// This class can generate string based code.
+class ProvidableVariableCodeGenerator {
```

Comment:
This looks good. But we should probably sync on one approach here.
So currently we have some object which has a similar naming pattern called `SourceCodeGenerator` and this returns code-strings. It does not write directly into the code writer.. 
And now your new `ProvidableVariableCodeGenerator` writes directly into the writer.

In the old code the `AnnotationVisitor` was doing this..
And you chnages it now so that the `ModelVisitor` returns data up to `widget_driver_generator` and there we grab that and pass it into the CodeGenerator..

But in the old logic the `widget_driver_generator` created a `AnnotationVisitor` which internally called the CodeGenerator..

I am not sure which approach is nicer.. But we should try to have one approach. 
So either also update the old code. Or make your new code behave a bit like the old..